cg/clients/freshdesk/freshdesk_client.py

```diff
@@ -11,28 +19,47 @@
 class FreshdeskClient:
     """Client for communicating with the freshdesk REST API."""
 
-    def __init__(self):
-        self.headers = None
-        self.api_key = None
-        self.url = None
-
-    def init_app(self, url: str, api_key: str):
-        """Set up the client."""
-        self.url = url
+    def __init__(self, base_url: str, api_key: str):
+        self.base_url = base_url
         self.api_key = api_key
-        self.headers = {"Content-Type": "application/json"}
-
-    @property
-    def auth_header(self):
-        return self.api_key, "X"
 
     def create_ticket(self, ticket: TicketCreate) -> TicketResponse:
         """Create a ticket."""
         LOG.debug(ticket.model_dump_json())
-        response = requests.post(
-            url=f"{self.url}{EndPoints.TICKETS}",
-            headers=self.headers,
-            auth=self.auth_header,
-            json=ticket.model_dump(exclude_none=True),
+        session: Session = self._get_session()
+        try:
+            response: Response = session.post(
+                url=f"{self.base_url}{EndPoints.TICKETS}",
+                json=ticket.model_dump(exclude_none=True),
+            )
+            response.raise_for_status()
+            ticket_response: TicketResponse = TicketResponse.model_validate(response.json())
+            return ticket_response
+        except RequestException as error:
+            LOG.error(f"Could not create ticket: {error}")
+            raise FreshdeskAPIException(error) from error
+        except ValidationError as error:
+            LOG.error(f"Response from Freshdesk does not fit model: {TicketResponse}.\n{error}")
+            raise FreshdeskModelException(error) from error
+        finally:
+            session.close()
+
+    def _get_session(self) -> Session:
+        """Configures and sets a session to be used for requests."""
+        session = Session()
+        self._configure_retries(session)
+        session.auth = (self.api_key, "X")
+        session.headers.update({"Content-Type": "application/json"})
+        return session
+
+    @staticmethod
+    def _configure_retries(session: Session) -> None:
+        """Configures retries for the session."""
+        retry_strategy = Retry(
+            total=5,
+            status_forcelist=[HTTPStatus.TOO_MANY_REQUESTS],
```

Comment:
Maybe we can use it for some more transient errors as well? Here is another library that I've used and the defaults they use: [default transient http errors](https://github.com/App-vNext/Polly/wiki/Polly-and-HttpClientFactory#extending-the-convenience-addtransienthttperrorpolicy-definition).

I think it makes sense to also, at least, use the retry for these errors
- 503 Service Unavailable
- 504 Gateway Timeout Error
- 408 Request Timeout Error

Reply:
Yes! ChatGPT recomended these: HTTPStatus.TOO_MANY_REQUESTS, HTTPStatus.INTERNAL_SERVER_ERROR, HTTPStatus.BAD_GATEWAY, HTTPStatus.SERVICE_UNAVAILABLE, HTTPStatus.GATEWAY_TIMEOUT. Will add 408 as well 👍